clconf/template/template_funcs.go

```diff
@@ -281,3 +284,87 @@ func RegexReplace(regex, src, repl string) (string, error) {
 	}
 	return re.ReplaceAllString(src, repl), nil
 }
+
+func sortType(input []string) (string, error) {
+	r := "string"
+	if len(input) > 0 {
+		r = input[0]
+	}
+	if r != "string" && r != "int" {
+		return "", fmt.Errorf("sort: Type '%s' is not supported (only int, string)", r)
+	}
+	return r, nil
+}
+
+// Sort sorts a copy of the input as specified type (string, int, default: string).
+func Sort(v []string, asType ...string) ([]string, error) {
+	sortType, err := sortType(asType)
+	if err != nil {
+		return nil, err
+	}
+
+	sortStrings := true
+	var ints []int
+	if sortType == "int" {
+		sortStrings = false
+		ints = make([]int, len(v))
+		for i, val := range v {
+			var err error
+			ints[i], err = strconv.Atoi(val)
+			if err != nil {
+				return nil, err
+			}
+		}
+		sort.Ints(ints)
+	}
+
+	r := make([]string, len(v))
+	for i, val := range v {
+		switch sortType {
+		case "int":
+			r[i] = fmt.Sprintf("%d", ints[i])
+		default:
+			r[i] = val
+		}
+	}
+	if sortStrings { // Don't sort if we've already sorted as ints
+		sort.Strings(r)
+	}
+	return r, nil
+}
+
+func getsvs(s *memkv.Store) func(string, ...string) ([]string, error) {
+	return func(pattern string, asType ...string) ([]string, error) {
+		vals, err := s.GetAllValues(pattern)
+		if err != nil {
+			return nil, err
+		}
+		return Sort(vals, asType...)
+	}
+}
+
+func getksvs(s *memkv.Store) func(string, ...string) ([]string, error) {
+	return func(pattern string, asType ...string) ([]string, error) {
+		ks, err := s.GetAll(pattern)
+		if err != nil {
+			return nil, err
+		}
+		kvMap := make(map[string]string)
+		keys := make([]string, len(ks))
+		r := make([]string, len(ks))
+		for i, kv := range ks {
+			key := kv.Key[strings.LastIndex(kv.Key, "/")+1:]
```

Comment:
how is this going to work for something like:

```yaml
/hip/hop: zip
/hip/foo/1: bar
/hip/foo/2: baz
```

I could be off in what is allowed as a result of pattern though, maybe it only allows for immediate children...

Reply:
Seems ok:

```$ go run . --pipe getv --template-string '{{getksvs "/foo/*"}}' <<EOF
foo:
  bar: 1
  stuff: good
  baz:
    key1: foobar
    key2: barfoo
EOF

[1 good]
```